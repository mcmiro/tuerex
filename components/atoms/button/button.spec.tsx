/* eslint-disable */
import * as React from "react";
import {screen, render} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Button from ".";

describe("Test Button component", () => {
	it("Should pass when type props and size props is defined", () => {
		render(
			React.cloneElement(
				<Button variant="text" data-testid="button-test">
					Test
				</Button>
			)
		);

		const button = screen.queryByTestId("button-test");
		expect(button).toBeDefined();
		expect(button).toMatchSnapshot();
	});

	it("Should pass when props are added", () => {
		render(
			React.cloneElement(
				<Button variant="contained" size="lg" disabled data-testid="button-test">
					Test
				</Button>
			)
		);

		const button = screen.queryByTestId("button-test");
		expect(button).toBeDefined();
		expect(button).toMatchSnapshot();
	});

	it("Should be called onClick for none disabled button", async () => {
		const mockClick = jest.fn();

		render(
			<Button data-testid="button-test" variant="contained" onClick={mockClick}>
				Test
			</Button>
		);

		const button = screen.queryByTestId("button-test");
		button && userEvent.click(button);
		button && expect(mockClick).toHaveBeenCalled();
	});

	it("Should not be called onClick for disabled button", () => {
		const mockClick = jest.fn();

		render(
			<Button data-testid="button-test" disabled variant="contained" onClick={mockClick}>
				Test
			</Button>
		);

		const button = screen.queryByTestId("button-test");
		button && userEvent.click(button);
		button && expect(mockClick).not.toHaveBeenCalled();
	});
});
