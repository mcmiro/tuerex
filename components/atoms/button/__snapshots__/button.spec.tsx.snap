// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component Should pass when props are added 1`] = `null`;

exports[`Test Button component Should pass when type props and size props is defined 1`] = `null`;
